File: core/src/main/java/com/graphhopper/routing/util/parsers/BikePriorityParser.java
package com.graphhopper.routing.util.parsers;

import com.graphhopper.routing.ev.DecimalEncodedValue;
import com.graphhopper.routing.ev.EncodedValueLookup;
import com.graphhopper.routing.ev.VehiclePriority;

import static com.graphhopper.routing.util.PriorityCode.BAD;

public class BikePriorityParser extends BikeCommonPriorityParser {

    public BikePriorityParser(EncodedValueLookup lookup) {
        this(lookup.getDecimalEncodedValue(VehiclePriority.key("bike")));
    }

    public BikePriorityParser(DecimalEncodedValue priorityEnc) {
        super(priorityEnc);

        addPushingSection("path");

        avoidHighwayTags.put("primary", BAD);
        avoidHighwayTags.put("primary_link", BAD);

        preferHighwayTags.add("service");
        preferHighwayTags.add("residential");
        preferHighwayTags.add("unclassified");
        preferHighwayTags.add("cycleway");

        setSpecificClassBicycle("touring");
    }
}

File: core/src/main/kotlin/com/graphhopper/routing/util/parsers/BikePriorityParser.kt
package com.graphhopper.routing.util.parsers

import com.graphhopper.routing.ev.DecimalEncodedValue
import com.graphhopper.routing.ev.EncodedValueLookup
import com.graphhopper.routing.ev.VehiclePriority
import com.graphhopper.routing.util.PriorityCode.BAD

open class BikePriorityParser(priorityEnc: DecimalEncodedValue) : BikeCommonPriorityParser(priorityEnc) {

    constructor(lookup: EncodedValueLookup) : this(lookup.getDecimalEncodedValue(VehiclePriority.key("bike")))

    init {
        addPushingSection("path")

        avoidHighwayTags["primary"] = BAD
        avoidHighwayTags["primary_link"] = BAD

        preferHighwayTags.add("service")
        preferHighwayTags.add("residential")
        preferHighwayTags.add("unclassified")
        preferHighwayTags.add("cycleway")

        setSpecificClassBicycle("touring")
    }
}
